lab4/viewpoints/context/context.tex

```diff
@@ -1,8 +1,220 @@
+    \item \textbf{Interaction Automation Level:}
+          \begin{itemize}
+              \item All operations are automated.
+          \end{itemize}
+    \item \textbf{Transactional Nature:}
+          \begin{itemize}
+              \item Interactions are transactional to ensure data integrity.
+          \end{itemize}
+    \item \textbf{Criticality and Timeliness:}
+          \begin{itemize}
+              \item High criticality for data retrieval and storage operations.
+              \item Timeliness is only somewhat important as users can endure some loading times.
+          \end{itemize}
+    \item \textbf{Interaction Mode:}
+          \begin{itemize}
+              \item Entirely message-based.
+          \end{itemize}
+    \item \textbf{Security Requirements:}
+          \begin{itemize}
+              \item High security required, including data encryption.
+          \end{itemize}
+    \item \textbf{Service Levels:}
+          \begin{itemize}
+              \item High availability, scalability, and performance required.
+          \end{itemize}
+    \item \textbf{Technical Nature:}
+          \begin{itemize}
+              \item Standard Object Storage System protocols.
+          \end{itemize}
+    \item \textbf{Data and File Formats:}
+          \begin{itemize}
+              \item Data is stored in large binary objects.
+          \end{itemize}
+\end{itemize}
+
+
+\subsection{Other External Interdependencies}
+
+No other external interdependencies are identified.
+
+
+\subsection{Impact of the System on Its Environment}
+
+\begin{itemize}
+    \item Improved learning management for students and teachers.
+    \item Potential reduction in the usage of other learning systems if EduPal is more efficient.
+    \item Students may migrate their conspectuses and other content to EduPal.
+\end{itemize}
+
+
+\subsection{Overall Completeness, Consistency, and Coherence}
+
+\begin{itemize}
+    \item Ensure all functionalities are well-integrated and user experiences are seamless.
+    \item Maintain data consistency across all modules and interfaces.
+    \item Plan for scalability and future integrations.
+\end{itemize}
+
+\subsection{Models}
+\subsection*{Context Diagram}
+
 \begin{figure}[H]
-  \includedrawio[width=\textwidth]{viewpoints/context/context.drawio}
-  \label{fig:viewpoint-context-main}
+    \includedrawio[width=\textwidth]{viewpoints/context/context.drawio}
+    \caption{Context diagram}
+    \label{context-view-context-diagram}
 \end{figure}
 
-test
+The context diagram visually represents EduPal as a central system interacting with its external entities.
+
+\begin{itemize}
+    \item \textbf{EduPal System:} Represents the main application.
+    \item \textbf{Users (Students and Teachers):} Connected to EduPal with interactions for uploading and accessing course information, using other EduPal features.
+    \item \textbf{Object storage system:} Connected to EduPal for conspectus file storage and retrieval.
+\end{itemize}
```

Comment:
Put diagram in beginning, no need to explain entities since other subsections do that.